### PR TITLE
Implement HTMLOptionElement named constructor

### LINKED ISSUE #25380: Implement "Option" named constructor for HTMLOptionElement
A constructor named Option is specified in https://html.spec.whatwg.org/#dom-option, expected in passing by various WPT tests, and specifically tested in html/semantics/forms/the-option-element/option-element-constructor.html. Servo has it commented out in HTMLOptionElement.webidl.

### PR DESCRIPTION
**Pull request by teapotd**
This change implements `Option` named constructor for `HTMLOptionElement`.

---
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #25380
- [X] There are tests for these changes

### CODE REVIEW

File: components/script/dom/htmloptionelement.rs [hunk 1]
```diff
@@ -72,6 +75,30 @@ impl HTMLOptionElement {
...
+    // https://html.spec.whatwg.org/multipage/#dom-option
+    pub fn Option(
+        window: &Window,
+        text: DOMString,
+        value: Option<DOMString>,
+        default_selected: bool,
+        selected: bool,
+    ) -> Fallible<DomRoot<HTMLOptionElement>> {
+        let document = window.Document();
+        let option = HTMLOptionElement::new(local_name!("option"), None, &document);
```
Comment (jdm) [inline components/script/dom/htmloptionelement.rs]: We should be using the `create_html_element` function from components/script/dom/create.rs instead, according to the specification.

**Review by jdm — CHANGES_REQUESTED** [review]
The rest the implementation looks sensible to me. Thanks for fixing this!

File: components/script/dom/htmloptionelement.rs (hunk 1 above)
Comment (teapotd) [reply components/script/dom/htmloptionelement.rs]: Looks like Image and Audio constructors should be updated to use `create_html_element` as well:

https://github.com/servo/servo/blob/1e0f2e59b7fe5238e42b53099cc992ca6d2ec4d3/components/script/dom/htmlimageelement.rs#L1254-L1260

https://github.com/servo/servo/blob/1e0f2e59b7fe5238e42b53099cc992ca6d2ec4d3/components/script/dom/htmlaudioelement.rs#L52-L54

https://html.spec.whatwg.org/multipage/embedded-content.html#dom-image
https://html.spec.whatwg.org/multipage/media.html#dom-audio

Comment (jdm) [reply components/script/dom/htmloptionelement.rs]: Indeed!

**Comment by jdm** [comment]
You will need to add Option to the list at https://github.com/servo/servo/blob/b3e6555bf4f02ae97dc2ab5e346a4382fef0c511/tests/wpt/mozilla/tests/mozilla/interfaces.html#L13 to avoid that failure.

**Comment by jdm** [comment]
```
+ ./mach test-tidy --no-progress --all
 0:07.99(B INFO(B Diffing old and new manifests /repo/tests/wpt/mozilla/meta/MANIFEST.json
 0:08.17(B WARNING(B Manifest /repo/tests/wpt/mozilla/meta/MANIFEST.json contains correct tests but file hashes changed.
 0:08.17(B ERROR(B Manifest /repo/tests/wpt/mozilla/meta/MANIFEST.json is outdated, use |./mach update-manifest| to fix.
 0:08.18(B INFO(B Diffing old and new manifests /repo/tests/wpt/webgl/meta/MANIFEST.json
 0:08.66(B INFO(B Diffing old and new manifests /repo/tests/wpt/metadata/MANIFEST.json
```
Whoops, this needs a `./mach update-manifest` since the test file was modified.

**Comment by teapotd** [comment]
Oh, sorry about that!

**Comment by jdm** [comment]
@bors-servo retry
* #24622

**Comment by jdm** [comment]
@bors-servo retry
* #24762

**Pull request merged**